components/pkg-export-tar/src/build.rs

```diff
@@ -0,0 +1,262 @@
+    ///
+    /// * If a temporary directory cannot be created
+    /// * If the root file system cannot be created
+    /// * If the `BuildRootContext` cannot be created
+    pub fn create(self, ui: &mut UI) -> Result<TempDir> {
+        let workdir = TempDir::new(&*PROGRAM_NAME)?;
+        let rootfs = workdir.path().join("rootfs");
+
+        ui.status(
+            Status::Creating,
+            format!("build root in {}", workdir.path().display()),
+        )?;
+        self.prepare_rootfs(ui, &rootfs)?;
+
+        Ok(workdir)
+    }
+
+    fn prepare_rootfs<P: AsRef<Path>>(&self, ui: &mut UI, rootfs: P) -> Result<()> {
+        ui.status(Status::Creating, "root filesystem")?;
+        if cfg!(target_os = "linux") {
+            rootfs::create(&rootfs)?;
+        }
+        self.create_symlink_to_artifact_cache(ui, &rootfs)?;
+        self.create_symlink_to_key_cache(ui, &rootfs)?;
+        self.install_base_pkgs(ui, &rootfs)?;
+        self.install_user_pkgs(ui, &rootfs)?;
+        self.remove_symlink_to_key_cache(ui, &rootfs)?;
+        self.remove_symlink_to_artifact_cache(ui, &rootfs)?;
+
+        Ok(())
+    }
+
+    fn create_symlink_to_artifact_cache<P: AsRef<Path>>(
+        &self,
+        ui: &mut UI,
+        rootfs: P,
+    ) -> Result<()> {
+        ui.status(Status::Creating, "artifact cache symlink")?;
+        let src = cache_artifact_path(None::<P>);
+        let dst = rootfs.as_ref().join(CACHE_ARTIFACT_PATH);
+        stdfs::create_dir_all(dst.parent().expect("parent directory exists"))?;
+        debug!(
+            "Symlinking src: {} to dst: {}",
+            src.display(),
+            dst.display()
+        );
+
+        Ok(symlink(src, dst)?)
+    }
+
+    fn create_symlink_to_key_cache<P: AsRef<Path>>(&self, ui: &mut UI, rootfs: P) -> Result<()> {
+        ui.status(Status::Creating, "key cache symlink")?;
+        let src = cache_key_path(None::<P>);
+        let dst = rootfs.as_ref().join(CACHE_KEY_PATH);
+        stdfs::create_dir_all(dst.parent().expect("parent directory exists"))?;
+        debug!(
+            "Symlinking src: {} to dst: {}",
+            src.display(),
+            dst.display()
+        );
+
+        Ok(symlink(src, dst)?)
+    }
+
+    fn install_base_pkgs<P: AsRef<Path>>(&self, ui: &mut UI, rootfs: P) -> Result<BasePkgIdents> {
+        let hab = self.install_base_pkg(ui, self.hab, &rootfs)?;
+        let sup = self.install_base_pkg(ui, self.hab_sup, &rootfs)?;
+        let launcher = self.install_base_pkg(ui, self.hab_launcher, &rootfs)?;
+        let busybox = if cfg!(target_os = "linux") {
+            Some(self.install_base_pkg(ui, BUSYBOX_IDENT, &rootfs)?)
+        } else {
+            None
+        };
+
+        Ok(BasePkgIdents {
+            hab,
+            sup,
+            launcher,
+            busybox,
+        })
+    }
+
+    fn install_user_pkgs<P: AsRef<Path>>(
+        &self,
+        ui: &mut UI,
+        rootfs: P,
+    ) -> Result<Vec<PackageIdent>> {
```

Comment:
Ditto w/r/t taking a single package for export.

Reply:
Fixed!